storey/table.py

```diff
@@ -235,11 +251,122 @@ def __getitem__(self, key):
+                    self._pending_by_key[job.key].pending = []
+
+                    future_task = self._safe_process_events(self._pending_by_key[job.key].in_flight)
+                    tail_position = received_job_count + self._q.qsize()
+                    jobs_at_tail = self_sent_jobs.get(tail_position, [])
+                    jobs_at_tail.append((job, asyncio.get_running_loop().create_task(future_task)))
+                    self_sent_jobs[tail_position] = jobs_at_tail
+                else:
+                    del self._pending_by_key[job.key]
+        except BaseException as ex:
+            if task and task is not _termination_obj and task[0].extra_data and task[0].extra_data._awaitable_result:
+                task[0].extra_data._awaitable_result._set_error(ex)
+            if not self._q.empty():
+                await self._q.get()
+            raise ex
+
+    async def _terminate(self):
+        if self._q:
+            self._terminated = True
+            last_flush = time.monotonic() - (self._flush_interval_milli * 1000)
+            for key, item in self._attrs_cache.items():
+                if item.last_change > last_flush:
+                    await self._persist(_PersistJob(key, None, None))
+
+            await self._q.put(_termination_obj)
+            await self._worker_awaitable
+            self._q = None
+
+    async def _persist(self, job):
+        if not self._q:
+            self._terminated = False
+            self._q = asyncio.queues.Queue(self._max_updates_in_flight)
+            self._worker_awaitable = asyncio.get_running_loop().create_task(self._persist_worker())
+
+        if self._worker_awaitable.done():
+            await self._worker_awaitable
+            raise FlowError("ConcurrentByKeyJobExecution worker has already terminated")
+        else:
+            # Initializing the key with 2 lists. One for pending requests and one for requests that an update request has been issued for.
+            if job.key not in self._pending_by_key:
+                self._pending_by_key[job.key] = _PendingEvent()
+
+            # If there is a current update in flight for the key, add the event to the pending list. Otherwise update the key.
+            self._pending_by_key[job.key].pending.append(job)
+            if len(self._pending_by_key[job.key].in_flight) == 0:
+                self._pending_by_key[job.key].in_flight = self._pending_by_key[job.key].pending
+                self._pending_by_key[job.key].pending = []
+                task = self._safe_process_events(self._pending_by_key[job.key].in_flight)
+                await self._q.put((job, asyncio.get_running_loop().create_task(task)))
+                if self._worker_awaitable.done():
+                    await self._worker_awaitable
+
+    async def _safe_process_events(self, jobs):
+        try:
+            # TODO only last??
+            job = jobs[-1]
```

Comment:
Yes, because they're supposed to have the same key?

Reply:
i kept the same behaviour as it was before (taking only the last key) but that's not entirely correct 
it will work fine for aggregations but not for static attrs.
each event may have different attrs to write and you could miss some if writing only the last one

Reply:
Decided to update the TODO with this information.